sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/utils/ResourceParser.java

```diff
@@ -26,45 +40,51 @@ public static void updateRoleNameAndInstance(
-                if (websiteSiteName != null) {
-                    serviceName = websiteSiteName;
-                }
-            }
-            String serviceNamespace = resource.getAttribute(ResourceAttributes.SERVICE_NAMESPACE);
-            String roleName = null;
-            if (serviceName != null && serviceNamespace != null) {
-                roleName = "[" + serviceNamespace + "]/" + serviceName;
-            } else if (serviceName != null) {
-                roleName = serviceName;
-            } else if (serviceNamespace != null) {
-                roleName = "[" + serviceNamespace + "]";
-            }
-            if (roleName != null) {
-                builder.addTag(ContextTagKeys.AI_CLOUD_ROLE.toString(), roleName);
+        builder.addTag(ContextTagKeys.AI_CLOUD_ROLE.toString(), getRoleName(resource));
+
+        builder.addTag(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(), getRoleInstance(resource));
+    }
+
+    private String getRoleName(Resource resource) {
+        String serviceName = resource.getAttribute(ResourceAttributes.SERVICE_NAME);
+        if (serviceName == null || DEFAULT_SERVICE_NAME.equals(serviceName)) {
+            if (websiteSiteName != null) {
+                serviceName = websiteSiteName;
             }
         }
+        String serviceNamespace = resource.getAttribute(ResourceAttributes.SERVICE_NAMESPACE);
+        if (serviceName != null && serviceNamespace != null) {
+            return "[" + serviceNamespace + "]/" + serviceName;
+        } else if (serviceName != null) {
+            return serviceName;
+        } else if (serviceNamespace != null) {
+            return "[" + serviceNamespace + "]";
+        }
+        return DEFAULT_SERVICE_NAME; // service.name is required so shouldn't get here anyways
+    }
 
-        if (existingTags == null
-            || !existingTags.containsKey(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString())) {
-            String roleInstance = resource.getAttribute(ResourceAttributes.SERVICE_INSTANCE_ID);
-            if (roleInstance == null) {
-                roleInstance = Strings.trimAndEmptyToNull(configProperties.getString("WEBSITE_INSTANCE_ID"));
-            }
-            if (roleInstance == null) {
-                roleInstance = HostName.get(); // default hostname
-            }
-            if (roleInstance != null) {
-                builder.addTag(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(), roleInstance);
-            }
+    private String getRoleInstance(Resource resource) {
+        String roleInstance = resource.getAttribute(ResourceAttributes.SERVICE_INSTANCE_ID);
+        if (roleInstance != null) {
+            return roleInstance;
+        }
+        roleInstance = websiteSiteInstance;
+        if (roleInstance != null) {
+            return roleInstance;
+        }
+        return HostName.get(); // default hostname
+    }
+
+    private static String getWebsiteSiteNameEnvVar(Map<String, String> envVars) {
```

Comment:
agent tooling in our agent has this exact code. maybe make it public so that it can be reused in our agent?